c/certificateService.c

```diff
@@ -118,36 +119,80 @@ static void respondWithBadRequest(HttpResponse *response) {
 
     jsonStart(p);
     {
-      jsonAddString(p, "error", "The length of the certificate is longer than 4096 bytes");
+      jsonAddString(p, "error", value);
     }
     jsonEnd(p);
 
     finishResponse(response);
 }
 
+
 static int serveMappingService(HttpService *service, HttpResponse *response)
 {
   HttpRequest *request = response->request;
 
   if (!strcmp(request->method, methodPOST))
   {
-    RUsermapParamList *userMapCertificateStructure 
-      = (RUsermapParamList*)safeMalloc31(sizeof(RUsermapParamList),"RUsermapParamList");
-    memset(userMapCertificateStructure, 0, sizeof(RUsermapParamList));
-
-    if(request->contentLength > sizeof(userMapCertificateStructure->certificate) || request->contentLength < 0) {
-      respondWithBadRequest(response);
-      return 0;
+    RUsermapParamList userMapStructure;
+
+    memset(&userMapStructure, 0, sizeof(RUsermapParamList));
+
+    char translatedURL[strlen(request->uri) +1];
```

Comment:
Should we limit this length? It doesn't feel right to allow users control a buffer size in memory via something they pass via web.

Reply:
is it ok if we return an error message in case of URI longer than 64? Or better to limit it to less?

Reply:
Yes, any user input must be validated and if it doesn't conform to the API requirements, an error is returned.